### PR TITLE
Testing | Applitools VRT | Base utils and accordion

### PR DESCRIPTION
**Pull request by joekarasek**
## Summary

This PR adds an integration with Applitools to Nightwatch.js.

## Details

There is a new Applitools class (singleton pattern to instantiate once). This can be imported into any nightwatch script. By running the function 
```
applitools.vrtTest({
      browser,
      testName: 'bolt-accordion',
      testingPath:
        '/pattern-lab/patterns/02-components-accordion-40-accordion-content-variations/02-components-accordion-40-accordion-content-variations.html',
    });
```
Nightwatch will take a full screenshot of the given url and upload that screenshot to Applitools.

## How to test

Log in to Applitools using Nedra's credentials. You should see multiple results for the Accordion component in different environments.

### CODE REVIEW

**Comment by margoromo** [comment]
Created http://vjira2:8080/browse/BDS-1711 for @sghoweri to review this.

**Comment by margoromo** [comment]
Created http://vjira2:8080/browse/BDS-1739 and http://vjira2:8080/browse/BDS-1740 for @danielamorse and @adamszalapski to review this as well.

**Comment by margoromo** [comment]
http://vjira2:8080/browse/BDS-1741 and http://vjira2:8080/browse/BDS-1742 created for @mikemai2awesome  and @remydenton

**Comment by adamszalapski** [comment]
@joekarasek Where i can find Nedra's credentials?

**Comment by adamszalapski** [comment]
Hi @joekarasek 
I try to run this locally but without success. 
I added Applitools function in Bolt Video Component but the first tests are not passing, second, i have no new screenshots on Applitools. Please let me know if this working only with Travis or locally too.

File: package.json [hunk 1]
```diff
@@ -98,24 +99,22 @@
     "extends @bolt/browserslist-config"
   ],
   "dependencies": {
-    "conf": "^5.0.0",
-    "get-port": "^5.0.0",
-    "patch-package": "^6.1.2",
-    "postinstall-postinstall": "^2.0.0",
+    "@bolt/testing-utils": "^2.5.2",
```
[the remark this answers is not in the document]
Comment (adamszalapski) [reply package.json]: @remydenton if we start using terminal commands to update `package.json` (add, remove packages) the alphabetical order will be done by command. We never should change those files manually.

[the remark this answers is not in the document]
Comment (remydenton) [reply package.json]: Yep, exactly @adamszalapski.  I guess the reminder is that _if_ you are taking shortcuts and updating them manually (as this file has been...), don't forget the importance of alphabetical order ;)

[the remark this answers is not in the document]
Comment (adamszalapski) [reply package.json]: I found perfect command for that use only `npx sort-package-json` in the folder with package.json file :D

File: docs-site/src/assets/images/band-shapes.svg [hunk 2]
```diff
@@ -1,36 +1 @@
...
-            </g>
-        </g>
-    </g>
-</svg>
\ No newline at end of file
+<svg width="1963" height="522" viewBox="0 0 1963 522" xmlns="http://www.w3.org/2000/svg"><g transform="translate(.873 .373)" fill="none" fill-rule="evenodd"><g transform="translate(1242 313)"><path fill="#E64B17" d="M208 0H0v208h208z"/><circle stroke="#1F2555" stroke-width="30" cx="104" cy="104" r="52"/></g><path d="M773.545 312.545v208h-208v-208h208zm-52 52h-104v104h104v-104z" fill="#E64B17"/><circle stroke="#FFCC4C" stroke-width="30" cx="877.545" cy="416.545" r="52"/><circle stroke="#E64B17" stroke-width="30" cx="1814" cy="416" r="52"/><path fill="#099" d="M305.545 311.545h208v208h-208z"/><path fill="#161B3D" d="M253.545 312.545h-208v208h208z"/><path fill="#FFCC4D" d="M149.545 343.006l-73.539 73.54 73.54 73.538 73.538-73.539z"/><path d="M1189.545 312.545v208h-208v-208h208zm-104 30.461l-73.539 73.54 73.54 73.538 73.538-73.539-73.539-73.539z" fill="#099"/><path fill="#161B3D" d="M1962 .545H0v312h1962z"/><g fill="#FED161"><path d="M1710 313h-208v208h208z"/></g></g></svg>
```
Comment (sghoweri) [inline docs-site/src/assets/images/band-shapes.svg]: We shouldn't be committing optimized SVG assets back to the docs site (source) folder!

File: docs-site/src/assets/images/bolt-logo--dark.svg [hunk 3]
```diff
@@ -1,35 +1 @@
...
...0-.966.784-1.75 1.75-1.75h36.008a1.75 1.75 0 0 1 1.516.875l16.66 28.876A3.5 3.5 0 0 1 193.901 85H142.75a1.75 1.75 0 0 1-1.75-1.75z" fill="#E1E2EB"/><path d="M85 2.75v31.5A1.75 1.75 0 0 1 83.25 36H29.338a1.75 1.75 0 0 1-1.516-2.625L42.459 8.004A14 14 0 0 1 54.586 1H83.25c.966 0 1.75.784 1.75 1.75z" fill="#F63"/><path d="M100.75 1h42.664a14 14 0 0 1 12.127 7.004l14.637 25.371A1.75 1.75 0 0 1 168.662 36H100.75A1.75 1.75 0 0 1 99 34.25V2.75c0-.966.784-1.75 1.75-1.75z" fill="#FFCC4C"/><path d="M42.46 175.996l-14.638-25.371A1.75 1.75 0 0 1 29.338 148H97.25c.966 0 1.75.784 1.75 1.75v31.5a1.75 1.75 0 0 1-1.75 1.75H54.586a14 14 0 0 1-12.127-7.004z" fill="#6C79D9"/><path d="M114.75 148h53.912a1.75 1.75 0 0 1 1.516 2.625l-14.637 25.371A14 14 0 0 1 143.414 183H114.75a1.75 1.75 0 0 1-1.75-1.75v-31.5c0-.966.784-1.75 1.75-1.75z" fill="#12B3B3"/><path d="M71 100.75v31.5a1.75 1.75 0 0 1-1.75 1.75H19.242a1.75 1.75 0 0 1-1.516-.875l-16.66-28.876A3.5 3.5 0 0 1 4.099 99H69.25c.966 0 1.75.784 1.75 1.75z" fill="#6C79D9"/><path d="M86.75 99h107.152a3.5 3.5 0 0 1 3.031 5.249l-16.659 28.876a1.75 1.75 0 0 1-1.516.875H86.75a1.75 1.75 0 0 1-1.75-1.75v-31.5c0-.966.784-1.75 1.75-1.75z" fill="#12B3B3"/></g></svg>
```
Comment (sghoweri) [inline docs-site/src/assets/images/bolt-logo--dark.svg]: Same as above -- we shouldn't be committing minified SVGs back to the source folder.

File: docs-site/src/assets/images/bolt-logo--light.svg [hunk 4]
```diff
@@ -1,35 +1 @@
...
...0-.966.784-1.75 1.75-1.75h36.008a1.75 1.75 0 0 1 1.516.875l16.66 28.876A3.5 3.5 0 0 1 193.901 85H142.75a1.75 1.75 0 0 1-1.75-1.75z" fill="#E1E2EB"/><path d="M85 2.75v31.5A1.75 1.75 0 0 1 83.25 36H29.338a1.75 1.75 0 0 1-1.516-2.625L42.459 8.004A14 14 0 0 1 54.586 1H83.25c.966 0 1.75.784 1.75 1.75z" fill="#F63"/><path d="M100.75 1h42.664a14 14 0 0 1 12.127 7.004l14.637 25.371A1.75 1.75 0 0 1 168.662 36H100.75A1.75 1.75 0 0 1 99 34.25V2.75c0-.966.784-1.75 1.75-1.75z" fill="#FFCC4C"/><path d="M42.46 175.996l-14.638-25.371A1.75 1.75 0 0 1 29.338 148H97.25c.966 0 1.75.784 1.75 1.75v31.5a1.75 1.75 0 0 1-1.75 1.75H54.586a14 14 0 0 1-12.127-7.004z" fill="#6C79D9"/><path d="M114.75 148h53.912a1.75 1.75 0 0 1 1.516 2.625l-14.637 25.371A14 14 0 0 1 143.414 183H114.75a1.75 1.75 0 0 1-1.75-1.75v-31.5c0-.966.784-1.75 1.75-1.75z" fill="#12B3B3"/><path d="M71 100.75v31.5a1.75 1.75 0 0 1-1.75 1.75H19.242a1.75 1.75 0 0 1-1.516-.875l-16.66-28.876A3.5 3.5 0 0 1 4.099 99H69.25c.966 0 1.75.784 1.75 1.75z" fill="#6C79D9"/><path d="M86.75 99h107.152a3.5 3.5 0 0 1 3.031 5.249l-16.659 28.876a1.75 1.75 0 0 1-1.516.875H86.75a1.75 1.75 0 0 1-1.75-1.75v-31.5c0-.966.784-1.75 1.75-1.75z" fill="#12B3B3"/></g></svg>
```
Comment (sghoweri) [inline docs-site/src/assets/images/bolt-logo--light.svg]: And this one

File: docs-site/src/assets/images/content/logos/logo-capgemini.svg [hunk 5]
```diff
@@ -1,6 +1 @@
...
...0 1.21 11.523 4.76 11.523 1.746 0 3.64-2.098 4.59-6.494.445 2.924 1.923 6.494 4.808 6.494 1.683 0 3.13-1.703 4.247-3.967.308 2.33.968 3.967 2.243 3.967 3.317 0 3.206-13.212 5.832-13.212 2.048 0 1.414 13.212 6.29 13.212 2.32 0 3.42-1.978 4.074-4.458.897 3.649 2.347 4.458 3.539 4.458.755 0 1.326-.266 2.074-1.158-3.77-1.636-3.453-7.588-3.453-11.433M54.962 35.456c-1.171 0-1.79-1.111-1.79-2.37 0-3.387 2.488-5.186 5.554-6.73-.14 7.407-1.861 9.1-3.764 9.1m15.23-24.01c1.14 0 1.793 1.047 1.68 2.553-.133 1.769-1.42 3.792-3.657 5.383-1.17-3.325-.3-7.936 1.978-7.936m36.123-5.008c1.087-.037 1.846-.97 1.85-2.08.004-1.11-.746-1.981-1.833-1.942-1.087.037-1.973.968-1.978 2.079-.005 1.11.873 1.98 1.961 1.943m27.022.563c.989-.035 1.796-.923 1.8-1.98.005-1.059-.794-1.887-1.785-1.852-.991.036-1.796.924-1.801 1.981-.004 1.058.796 1.887 1.786 1.851m36.734 8.792c-.081-4.187-2.074-7.72-5.145-10.54-2.332-2.13-5.103-3.747-8.008-4.965a28.877 28.877 0 0 0-.705-.287h-.001c-3.577 4.285-15.963 7.483-15.963 16.467 0 3.512 2.218 6.796 5.475 8.114 1.888.714 3.777.753 5.667.118 1.68-.552 3.062-1.594 4.211-2.864 3.638-4.022 5.118-10.26 9.581-10.26 4.084 0 4.634 2.907 4.895 4.481 0-.004-.002-.114-.007-.264"/></g></svg>
```
Comment (sghoweri) [inline docs-site/src/assets/images/content/logos/logo-capgemini.svg]: And this one

File: docs-site/src/assets/images/content/logos/logo-paypal.svg [hunk 6]
```diff
@@ -1,55 +1 @@
...
....869-2.458 1.109-.906.236-1.939.355-3.072.355h-.73c-.522 0-1.029.188-1.427.525a2.21 2.21 0 0 0-.744 1.328l-.055.299-.924 5.855-.042.215c-.011.068-.03.102-.058.125a.155.155 0 0 1-.096.035H7.266z"/><path fill="#179BD7" d="M23.048 7.667c-.028.179-.06.362-.096.55-1.237 6.351-5.469 8.545-10.874 8.545H9.326c-.661 0-1.218.48-1.321 1.132L6.596 26.83l-.399 2.533a.704.704 0 0 0 .695.814h4.881c.578 0 1.069-.42 1.16-.99l.048-.248.919-5.832.059-.32c.09-.572.582-.992 1.16-.992h.73c4.729 0 8.431-1.92 9.513-7.476.452-2.321.218-4.259-.978-5.622a4.667 4.667 0 0 0-1.336-1.03z"/><path fill="#222D65" d="M21.754 7.151a9.757 9.757 0 0 0-1.203-.267 15.284 15.284 0 0 0-2.426-.177h-7.352a1.172 1.172 0 0 0-1.159.992L8.05 17.605l-.045.289a1.336 1.336 0 0 1 1.321-1.132h2.752c5.405 0 9.637-2.195 10.874-8.545.037-.188.068-.371.096-.55a6.594 6.594 0 0 0-1.017-.429 9.045 9.045 0 0 0-.277-.087z"/><path fill="#253B80" d="M9.614 7.699a1.169 1.169 0 0 1 1.159-.991h7.352c.871 0 1.684.057 2.426.177a9.757 9.757 0 0 1 1.481.353c.365.121.704.264 1.017.429.368-2.347-.003-3.945-1.272-5.392C20.378.682 17.853 0 14.622 0h-9.38c-.66 0-1.223.48-1.325 1.133L.01 25.898a.806.806 0 0 0 .795.932h5.791l1.454-9.225 1.564-9.906z"/></svg>
```
Comment (sghoweri) [inline docs-site/src/assets/images/content/logos/logo-paypal.svg]: And this one

File: docs-site/src/assets/images/docs/bolt-logo-lockup-dark.svg [hunk 7]
```diff
@@ -1 +1 @@
...
...4V.234c0-.13.105-.234.234-.234zM44.742 6.613c1.699 0 3.242.392 4.631 1.177 1.39.784 2.491 1.91 3.306 3.375.815 1.466 1.222 3.168 1.222 5.107 0 1.938-.407 3.646-1.222 5.123-.815 1.478-1.917 2.614-3.306 3.41-1.389.797-2.932 1.195-4.631 1.195-2.456 0-4.293-.727-5.51-2.181v1.635c0 .13-.104.234-.234.234h-5.764a.234.234 0 0 1-.234-.234V.234c0-.13.105-.234.234-.234h6.074c.13 0 .234.105.234.234v8.317c1.263-1.292 2.996-1.938 5.2-1.938zm-1.377 14.16c1.147 0 2.089-.399 2.823-1.195.735-.796 1.102-1.898 1.102-3.306 0-1.385-.367-2.47-1.102-3.255-.734-.784-1.676-1.177-2.823-1.177-1.148 0-2.09.393-2.824 1.177-.735.785-1.102 1.87-1.102 3.255 0 1.408.367 2.51 1.102 3.306.735.796 1.676 1.194 2.824 1.194z"/><path d="M62.806 26c-5.4 0-9.779-4.378-9.779-9.78 0-5.4 4.378-9.779 9.78-9.779 5.4 0 9.779 4.379 9.779 9.78 0 5.4-4.379 9.779-9.78 9.779zm-.058-5.27a4.45 4.45 0 1 0 0-8.901 4.45 4.45 0 0 0 0 8.9zM89.559 8.188l5.386-.016c.13 0 .235.104.235.234v5.302c0 .13-.104.234-.233.235l-5.388.016v4.785c0 1.035.839 1.874 1.873 1.874h3.514c.13 0 .234.104.234.234v4.919c0 .129-.105.234-.234.234h-5.387A6.559 6.559 0 0 1 83 19.446V6.596c0-.165.088-.318.23-.403l5.975-3.52a.234.234 0 0 1 .354.201v5.314z"/></g></g></svg>
```
Comment (sghoweri) [inline docs-site/src/assets/images/docs/bolt-logo-lockup-dark.svg]: And this one...

File: docs-site/src/assets/images/docs/bolt-logo.svg [hunk 8]
```diff
@@ -1 +1 @@
...
...0-.966.784-1.75 1.75-1.75h36.008a1.75 1.75 0 0 1 1.516.875l16.66 28.876A3.5 3.5 0 0 1 193.901 85H142.75a1.75 1.75 0 0 1-1.75-1.75z" fill="#e1e2eb"/><path d="M85 2.75v31.5A1.75 1.75 0 0 1 83.25 36H29.338a1.75 1.75 0 0 1-1.516-2.625L42.459 8.004A14 14 0 0 1 54.586 1H83.25c.966 0 1.75.784 1.75 1.75z" fill="#f63"/><path d="M100.75 1h42.664a14 14 0 0 1 12.127 7.004l14.637 25.371A1.75 1.75 0 0 1 168.662 36H100.75A1.75 1.75 0 0 1 99 34.25V2.75c0-.966.784-1.75 1.75-1.75z" fill="#ffcc4c"/><path d="M42.46 175.996l-14.638-25.371A1.75 1.75 0 0 1 29.338 148H97.25c.966 0 1.75.784 1.75 1.75v31.5a1.75 1.75 0 0 1-1.75 1.75H54.586a14 14 0 0 1-12.127-7.004z" fill="#6c79d9"/><path d="M114.75 148h53.912a1.75 1.75 0 0 1 1.516 2.625l-14.637 25.371A14 14 0 0 1 143.414 183H114.75a1.75 1.75 0 0 1-1.75-1.75v-31.5c0-.966.784-1.75 1.75-1.75z" fill="#12b3b3"/><path d="M71 100.75v31.5a1.75 1.75 0 0 1-1.75 1.75H19.242a1.75 1.75 0 0 1-1.516-.875l-16.66-28.876A3.5 3.5 0 0 1 4.099 99H69.25c.966 0 1.75.784 1.75 1.75z" fill="#6c79d9"/><path d="M86.75 99h107.152a3.5 3.5 0 0 1 3.031 5.249l-16.659 28.876a1.75 1.75 0 0 1-1.516.875H86.75a1.75 1.75 0 0 1-1.75-1.75v-31.5c0-.966.784-1.75 1.75-1.75z" fill="#12b3b3"/></g></svg>
```
Comment (sghoweri) [inline docs-site/src/assets/images/docs/bolt-logo.svg]: And this one...

File: packages/testing/testing-nightwatch/nightwatch.js [hunk 9]
```diff
@@ -1,5 +1,6 @@
 const globby = require('globby');
 const path = require('path');
+const uuidv1 = require('uuid/v1');
```
Comment (sghoweri) [inline packages/testing/testing-nightwatch/nightwatch.js]: @joekarasek lol I was just looking at this exact same package for internally having each Bolt web component instance internally assigned itself a unique id 🤓

File: packages/testing/testing-nightwatch/nightwatch.local.js [hunk 10]
```diff
@@ -24,6 +25,7 @@ module.exports = {
   globals: {
     waitForConditionTimeout: 5000,
     testingUrl,
+    applitoolsBatchId: process.env.TRAVIS_JOB_ID || '1-2-3-4-local-id', // @todo figure out how to set a uuid that persists across multiple tests. Nightwatch reboots per .e2e.js file
```
Comment (sghoweri) [inline packages/testing/testing-nightwatch/nightwatch.local.js]: @joekarasek @remydenton @adamszalapski who's plate is this TODO on? 😉

**Comment by sghoweri** [comment]
> Very promising, and thanks again for the demo @joekarasek! In addition to my one inline comment about `testMatchLevel`:
> 
> * To piggy back on @adamszalapski's request, I think it's worthwhile to have testing instructions that describe how to actually create and run a test of one's own.
> * Along those lines, this PR should include at least brief documentation of Applitools tests (probably next to or in https://boltdesignsystem.com/docs/development/bolt-testing-101.html).  Consider adding:
>   
>   * very brief overview of Applitools
>   * basics for how/where to write a new Applitools test
>   * any obvious best practices (e.g. when to use and not use this kind of test vs. another test type, how and when to change `testMatchLevel`, etc.)

@remydenton 100% agree with these!

**Comment by colbytcook** [comment]
Closing because PR is outdated

**Pull request closed without merging**